src/component/fiber.ts

```diff
@@ -237,7 +237,15 @@ export class Fiber {
         component.__patch(target!, fiber.vnode!);
       } else {
         if (fiber.shouldPatch) {
+          const _elm = component.__owl__.vnode!.elm;
           component.__patch(component.__owl__.vnode!, fiber.vnode!);
+          // When updating a Component's props (in directive),
+          // the component has a pvnode AND should be patched.
+          // However, its pvnode.elm may have changed if it is "virtual"
+          // i.e. coming from the child.
+          if (component.__owl__.pvnode && _elm !== component.__owl__.vnode!.elm) {
```

Comment:
can't you get rid of the second part of the condition (and always update the pvnode's elm) ?

Reply:
yes, I suppose I can, but there is a catch. I executed many times the test full test suite and on my branch, one test seem to crash undeterministicly. The second part of the condition seems to avoid that